isis/src/control/apps/cnetedit/cnetedit.cpp

```diff
@@ -0,0 +1,1127 @@
+
+            // Delete the validator
+            if (cneteditValidator != NULL) {
+                delete cneteditValidator;
+                cneteditValidator = NULL;
+            }
+
+            // Log the DEFFILE to the print file
+            results.addLogGroup(defFile.findGroup("ValidMeasure", Pvl::Traverse));
+        }
+    }
+
+    if (ui.GetBoolean("LOCK")) {
+        if (processPoints) lockPoints(cnet, *cpList);
+        if (processCubes) lockCubes(cnet, *cubeSnl);
+        if (processMeasures) lockMeasures(cnet, *editMeasuresList);
+    }
+
+    // Log statistics
+    if (keepLog) {
+        Pvl outputLog;
+
+        outputLog.addKeyword(PvlKeyword("PointsDeleted", toString(numPointsDeleted)));
+        outputLog.addKeyword(PvlKeyword("MeasuresDeleted", toString(numMeasuresDeleted)));
+
+        PvlObject lockedLog = createLog(
+            "EditLocked", editLockedPoints, editLockedMeasures);
+        outputLog.addObject(lockedLog);
+
+        outputLog.addObject(createLog("RetainedReferences", retainedReferences));
+
+        // Depending on whether the user chose to delete ignored points and
+        // measures, the log will either contain reasons for being ignored, or
+        // reasons for being deleted
+        PvlObject ignoredLog = createLog(
+            deleteIgnored ? "Deleted" : "Ignored", ignoredPoints, ignoredMeasures);
+        outputLog.addObject(ignoredLog);
+
+        // Write the log
+        QString logFileName = ui.GetFileName("LOG");
+        outputLog.write(logFileName);
+
+        // Delete the structures keeping track of the ignored points and measures
+        delete ignoredPoints;
+        ignoredPoints = NULL;
+
+        delete ignoredMeasures;
+        ignoredMeasures = NULL;
+
+        delete retainedReferences;
+        retainedReferences = NULL;
+
+        delete editLockedPoints;
+        editLockedPoints = NULL;
+
+        delete editLockedMeasures;
+        editLockedMeasures = NULL;
+    }
+
+    // Write the network
+    cnet.Write(ui.GetFileName("ONET"));
+
+    return results;
+  }
+
+
+  /**
+   * After any modification to a point's measures or ignored status, this check
+   * should be performed to determine if the changes should result in the point's
+   * deletion.
+   *
+   * @param point The Control Point recently modified
+   *
+   * @return Whether or not the point should be deleted
+   */
+  bool shouldDelete(ControlPoint *point){
+
+      if(!deleteIgnored)
+          return false;
+
+      else{
+
+          if ( point->GetNumMeasures() == 0 && !preservePoints)
+              return true;
+
+
+          if (point->GetType() != ControlPoint::Fixed && point->GetNumMeasures()< 2 ){
+
+              if(preservePoints  && point->GetNumMeasures() == 1)
+                  return false;
+
+              return true;  //deleteIgnore = TRUE or else we would not be here
+          }
+
+
+          if( point->IsIgnored() )
+              return true;
+
+          return false;
+      }
+  }
```

Comment:
minor nitpick, but the spacing here is very weird.

Reply:
Fixed.